test/parallel/test-dotenv-edge-cases.js

```diff
@@ -112,4 +112,32 @@ describe('.env supports edge cases', () => {
     assert.strictEqual(child.stderr, '');
     assert.strictEqual(child.code, 0);
   });
+
+  // Ref: https://github.com/nodejs/node/issues/54255
+  it('should handle when an unrelated argument starts with --env-file', async () => {
+    const child = await common.spawnPromisified(
+      process.execPath,
+      [
+        '--env-file-XYZ', validEnvFilePath,
+      ],
+      { cwd: fixtures.path('dotenv') },
+    );
+    assert.strictEqual(child.stdout, '');
+    assert.match(child.stderr, /bad option: --env-file-XYZ/);
```

Comment:
Please add a C++ test. There is no need to spawn a process for each input for testing a single function.

Reply:
Ahh okay, apologies for the misunderstanding.